contracts/middle-layer/resource-mirror/AdditionalBucketHub.sol

```diff
@@ -72,49 +72,144 @@ contract AdditionalBucketHub is BucketStorage, GnfdAccessControl {
      * @param synPkg Package containing information of the bucket to be created
      */
     function createBucket(CreateBucketSynPackage memory synPkg) external payable returns (bool) {
+        (uint8 _channelId, bytes memory _msgBytes, uint256 _relayFee, uint256 _ackRelayFee, ) = _prepareCreateBucket(
+            msg.sender,
+            synPkg
+        );
+        ICrossChain(CROSS_CHAIN).sendSynPackage(_channelId, _msgBytes, _relayFee, _ackRelayFee);
+        return true;
+    }
+
+    function prepareCreateBucket(
+        address sender,
+        CreateBucketSynPackage memory synPkg
+    ) external payable onlyMultiMessage returns (uint8, bytes memory, uint256, uint256, address) {
+        return _prepareCreateBucket(sender, synPkg);
+    }
+
+    /**
+     * @dev create a bucket and send cross-chain request from BSC to GNFD.
+     * Callback function will be called when the request is processed.
+     *
+     * @param synPkg Package containing information of the bucket to be created
+     * @param callbackGasLimit The gas limit for callback function
+     * @param extraData Extra data for callback function. The `appAddress` in `extraData` will be ignored.
+     * It will be reset to the `msg.sender` all the time. And make sure the `refundAddress` is payable.
+     */
+    function createBucket(
+        CreateBucketSynPackage memory synPkg,
+        uint256 callbackGasLimit,
+        ExtraData memory extraData
+    ) external payable returns (bool) {
+        (uint8 _channelId, bytes memory _msgBytes, uint256 _relayFee, uint256 _ackRelayFee, ) = _prepareCreateBucket(
+            msg.sender,
+            synPkg,
+            callbackGasLimit,
+            extraData
+        );
+
+        ICrossChain(CROSS_CHAIN).sendSynPackage(_channelId, _msgBytes, _relayFee, _ackRelayFee);
+        return true;
+    }
+
+    function prepareCreateBucket(
```

Comment:
We know that `AdditionalBucketHub.sol` is not the really contract, it is the implementation contract, the real contract is the `BucketHub`, so any newly added external call in this contract, should be added in `BucketHub`.

So is other resource contracts.

Reply:
fixed